### PR TITLE
Chapter 3

### PR DESCRIPTION
**Pull request by nuriman1**

### CODE REVIEW

**Comment by nuriman1** [comment]
ready

**Comment by thaodtd-0665** [comment]
ready

File: app/assets/stylesheets/static_pages.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the StaticPages controller here.
```
Comment (thaodtd-0665) [inline app/assets/stylesheets/static_pages.scss]: remove this file because it not use

File: app/views/static_pages/about.html.erb [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    The <a href="https://www.railstutorial.org/"><em>Ruby on Rails
+    Tutorial</em></a>, part of the
+    <a href="https://www.learnenough.com/">Learn Enough</a> family of
+    tutorials, is a
+    <a href="https://www.railstutorial.org/book">book</a> and
+    <a href="https://screencasts.railstutorial.org/">screencast series</a>
+    to teach web development with
+    <a href="https://rubyonrails.org/">Ruby on Rails</a>.
+    This is the sample app for the tutorial.
+</p>
```
Comment (thaodtd-0665) [inline app/views/static_pages/about.html.erb]: end line error must have an empty line at the end of file

**Comment by thaodtd-0665** [comment]
commented

**Comment by nuriman1** [comment]
ready

File: app/views/static_pages/about.html.erb (hunk 2 above)
Comment (nuriman1) [reply app/views/static_pages/about.html.erb]: ok done :)

File: app/assets/stylesheets/static_pages.scss (hunk 1 above)
Comment (nuriman1) [reply app/assets/stylesheets/static_pages.scss]: noted

File: app/controllers/static_pages_controller.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class StaticPagesController < ApplicationController
+  def home
+  end
```
Comment (vanvtt-0952) [inline app/controllers/static_pages_controller.rb]: change line number 2-3 
`def home`
`end`

to one line
`def home; end`

File: app/helpers/static_pages_helper.rb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (vanvtt-0952) [inline app/helpers/static_pages_helper.rb]: delete this file

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+    def full_title page_title = ""
+        base_title = "Ruby on Rails Tutorial Sample App"
+        page_title.empty? ? base_title : "#{page_title} | #{base_title}"
```
Comment (vanvtt-0952) [inline app/helpers/application_helper.rb]: use I18n for ` "Ruby on Rails Tutorial Sample App"` 
link: https://guides.rubyonrails.org/i18n.html

**Comment by thaodtd-0665** [comment]
ready

**Review by chungpv-1008 — COMMENTED** [review]
Not fixed comments yet

File: app/controllers/static_pages_controller.rb (hunk 3 above)
Comment (nuriman1) [reply app/controllers/static_pages_controller.rb]: ok done :)

File: app/helpers/static_pages_helper.rb (hunk 4 above)
Comment (nuriman1) [reply app/helpers/static_pages_helper.rb]: done

File: app/helpers/application_helper.rb (hunk 5 above)
Comment (nuriman1) [reply app/helpers/application_helper.rb]: ok done

**Comment by nuriman1** [comment]
ready

File: config/routes.rb [hunk 6]
```diff
@@ -1,3 +1,7 @@
 Rails.application.routes.draw do
+  root 'static_pages#home'
```
Comment (ngocvt-0484) [inline config/routes.rb]: use "

File: app/views/static_pages/home.html.erb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+<%= provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+    This is the home page for the
+    <a href="http://www.railstutorial.org/">Ruby on
```
Comment (ngocvt-0484) [inline app/views/static_pages/home.html.erb]: use link_to hepler

File: app/helpers/application_helper.rb (hunk 5 above)
Comment (ngocvt-0484) [reply app/helpers/application_helper.rb]: please tab 2 space

**Comment by nuriman1** [comment]
ready

File: Gemfile.lock [hunk 8]
```diff
@@ -73,7 +73,53 @@ GEM
       xpath (~> 3.2)
     childprocess (3.0.0)
     concurrent-ruby (1.1.7)
+<<<<<<< HEAD
     crass (1.0.6)
+=======
```
Comment (saobv-2223) [inline Gemfile.lock]: resolve conflict, please

File: app/helpers/application_helper.rb [hunk 9]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+    def full_title page_title = ""
+        base_title = t "user.welcome"
+        page_title.empty? ? base_title : "#{page_title} | #{base_title}"
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]:
```suggestion
      base_title = t "user.welcome"
      page_title.blank? ? base_title : "#{page_title} | #{base_title}"
```

File: app/views/layouts/application.html.erb [hunk 10]
```diff
@@ -2,6 +2,7 @@
 <html>
   <head>
     <title>SampleApp</title>
+    <title><%= yield(:title) %> | Ruby on Rails Tutorial Sample App</title>
```
Comment (saobv-2223) [inline app/views/layouts/application.html.erb]: remove ()

File: app/views/static_pages/about.html.erb [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+<p>
+    The <%= link_to "<em>Ruby on Rails
+    Tutorial</em>", "https://www.railstutorial.org/" %>, part of the
+    <%= link_to "Learn Enough", "https://www.learnenough.com/" %> family of
+    tutorials, is a
+    <%= link_to "book", "https://www.railstutorial.org/book" %> and
+    <%= link_to "screencast series", "https://screencasts.railstutorial.org/" %>
+    to teach web development with
+    <%= link_to "Ruby on Rails", "https://rubyonrails.org/" %>.
+    This is the sample app for the tutorial.
```
Comment (saobv-2223) [inline app/views/static_pages/about.html.erb]: image

File: app/views/static_pages/help.html.erb [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+
+<h1>Help</h1>
+<p>
+    Get help on the Ruby on Rails Tutorial at the
+    <%= link_to "Rails Tutorial help page", 
+    Settings.links.rails_tutorial.help_page %>
+    To get help on this sample app, see the
+    <%= link_to "<em>Ruby on Rails
+    Tutorial</em>book",
+    Settings.links.rails_tutorial.book_page %>
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb]: image

File: app/views/static_pages/home.html.erb [hunk 13]
```diff
@@ -0,0 +1,9 @@
+<%= provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+    This is the home page for the
+    <%= link_to "Ruby on Rails Tutorial", 
+    Settings.links.rails_tutorial.home_page %>
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]: image

File: config/settings.yml [hunk 14]
```diff
@@ -0,0 +1,5 @@
+links:
+  rails_tutorial:
+    home_page: "http://www.railstutorial.org/book"
+    help_page: "http://www.railstutorial.org/help"
+    book_page: "http://www.railstutorial.org/book"
```
Comment (saobv-2223) [inline config/settings.yml]: image

File: app/helpers/application_helper.rb (hunk 9 above)
Comment (nuriman1) [reply app/helpers/application_helper.rb]: ok done

File: app/views/layouts/application.html.erb (hunk 10 above)
Comment (nuriman1) [reply app/views/layouts/application.html.erb]: done

File: app/views/static_pages/about.html.erb (hunk 11 above)
Comment (nuriman1) [reply app/views/static_pages/about.html.erb]: done

File: app/views/static_pages/help.html.erb (hunk 12 above)
Comment (nuriman1) [reply app/views/static_pages/help.html.erb]: done

File: app/views/static_pages/home.html.erb (hunk 13 above)
Comment (nuriman1) [reply app/views/static_pages/home.html.erb]: done

File: config/settings.yml (hunk 14 above)
Comment (nuriman1) [reply config/settings.yml]: done

File: Gemfile.lock (hunk 8 above)
Comment (nuriman1) [reply Gemfile.lock]: done

File: app/views/layouts/application.html.erb (hunk 10 above)
Comment (nuriman1) [reply app/views/layouts/application.html.erb]: but after I removed the bracket, when I go to localhost:3000 it stated syntax error for that line

Comment (nuriman1) [reply app/views/layouts/application.html.erb]: oh solved already, I forgot the space

**Comment by nuriman1** [comment]
ready

File: app/helpers/application_helper.rb [hunk 15]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+    def full_title page_title = ""
+      base_title = t "user.welcome"
+      page_title.blank? ? base_title : "#{page_title} | #{base_title}"
+    end
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]:
```suggestion
  def full_title page_title = ""
    base_title = t "user.welcome"
    page_title.blank? ? base_title : "#{page_title} | #{base_title}"
  end
```

File: app/views/static_pages/about.html.erb [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+      The <%= link_to "<em>Ruby on Rails
+    Tutorial</em>", "https://www.railstutorial.org/" %>, part of the
+      <%= link_to "Learn Enough", "https://www.learnenough.com/" %> family of
+    tutorials, is a
+      <%= link_to "book", "https://www.railstutorial.org/book" %> and
+      <%= link_to "screencast series", "https://screencasts.railstutorial.org/" %>
+    to teach web development with
+      <%= link_to "Ruby on Rails", "https://rubyonrails.org/" %>.
+    This is the sample app for the tutorial.
+</p>
```
Comment (saobv-2223) [inline app/views/static_pages/about.html.erb]: image

File: app/views/static_pages/help.html.erb [hunk 17]
```diff
@@ -0,0 +1,15 @@
...
+    Tutorial at the
+      <%= link_to "Rails Tutorial help 
+    page", 
+    Settings.links.rails_tutorial.help_page %>
+      To get help on this sample app, 
+    see the
+      <%= link_to "<em>Ruby on Rails
+    Tutorial</em>book",
+    Settings.links.rails_tutorial.book_page %>
+</p>
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb]: image

File: app/views/static_pages/home.html.erb [hunk 18]
```diff
@@ -0,0 +1,10 @@
+<%= provide :title, "Home" %>
+
+<h1>Sample App</h1>
+<p>
+      This is the home page 
+    for the
+      <%= link_to "Ruby on Rails Tutorial", 
+    Settings.links.rails_tutorial.home_page %>
+    sample application.
+</p>
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]: image

File: config/application.rb [hunk 19]
```diff
@@ -15,5 +15,10 @@ class Application < Rails::Application
     # Application configuration can go into files in config/initializers
     # -- all .rb files in that directory are automatically loaded after loading
     # the framework and any gems in your application.
+<<<<<<< HEAD
+=======
+    config.i18n.available_locales = :en
+    config.i18n.default_locale = :en
+>>>>>>> Init Project
```
Comment (saobv-2223) [inline config/application.rb]: remove conflict

File: config/settings.yml [hunk 20]
```diff
@@ -0,0 +1,6 @@
...
+
```
Comment (saobv-2223) [inline config/settings.yml]: image

File: app/helpers/application_helper.rb (hunk 9 above)
Comment (nuriman1) [reply app/helpers/application_helper.rb]: done

File: app/helpers/application_helper.rb (hunk 15 above)
Comment (nuriman1) [reply app/helpers/application_helper.rb]: done

File: app/views/static_pages/about.html.erb (hunk 16 above)
Comment (nuriman1) [reply app/views/static_pages/about.html.erb]: done

File: app/views/static_pages/help.html.erb (hunk 17 above)
Comment (nuriman1) [reply app/views/static_pages/help.html.erb]: done

File: app/views/static_pages/home.html.erb (hunk 18 above)
Comment (nuriman1) [reply app/views/static_pages/home.html.erb]: done

File: config/application.rb (hunk 19 above)
Comment (nuriman1) [reply config/application.rb]: done

File: config/settings.yml (hunk 20 above)
Comment (nuriman1) [reply config/settings.yml]: i already put an end line at line 6

**Comment by nuriman1** [comment]
ready

File: app/helpers/application_helper.rb [hunk 21]
```diff
@@ -1,2 +1,6 @@
 module ApplicationHelper
+  def full_title page_title = ""
```
Comment (saobv-2223) [inline app/helpers/application_helper.rb]:
```suggestion
  def full_title page_title
```

File: app/views/static_pages/about.html.erb [hunk 22]
```diff
@@ -0,0 +1,12 @@
...
+
+<h1>About</h1>
+  <p>
+    <%= link_to "<em>Ruby on Rails Tutorial</em>", "https://www.railstutorial.org/" %>, part of the
+      <%= link_to "Learn Enough", "https://www.learnenough.com/" %> family of tutorials, is a
+        <%= link_to "book", "https://www.railstutorial.org/book" %> and
+          <%= link_to "screencast series", "https://screencasts.railstutorial.org/" %> 
+        to teach web development with
+      <%= link_to "Ruby on Rails", "https://rubyonrails.org/" %>.
+    This is the sample app for the tutorial.
```
Comment (saobv-2223) [inline app/views/static_pages/about.html.erb]: please, check indenting for those lines

File: app/views/static_pages/help.html.erb [hunk 23]
```diff
@@ -0,0 +1,10 @@
+<%= provide :title, "Help" %>
+
+<h1>Help</h1>
+  <p>
+    Get help on the Ruby on Rails Tutorial at the 
+     <%= link_to "Rails Tutorial help page", Settings.links.rails_tutorial.help_page %>
+      
+     To get help on this sample app, see the
+    <%= link_to "<em>Ruby on Rails Tutorial</em>book", Settings.links.rails_tutorial.book_page %>
+  </p>
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb]: remove 2 spaces

File: app/views/static_pages/home.html.erb [hunk 24]
```diff
@@ -0,0 +1,8 @@
+<%= provide :title, "Home" %>
+
+<h1>Sample App</h1>
+  <p>
+    This is the home page for the
+    <%= link_to "Ruby on Rails Tutorial", Settings.links.rails_tutorial.home_page %>
+    sample application.
+  </p>
```
Comment (saobv-2223) [inline app/views/static_pages/home.html.erb]: remove 2 spaces

File: config/settings.yml (hunk 20 above)
Comment (saobv-2223) [reply config/settings.yml]: Dont write anything at line 6, even if space

File: app/helpers/application_helper.rb (hunk 21 above)
Comment (nuriman1) [reply app/helpers/application_helper.rb]: done

File: config/settings.yml (hunk 20 above)
Comment (nuriman1) [reply config/settings.yml]: oh i see

Comment (nuriman1) [reply config/settings.yml]: can you check again if I do it correctly ?

File: app/views/static_pages/home.html.erb (hunk 24 above)
Comment (nuriman1) [reply app/views/static_pages/home.html.erb]: done but can you check again if I do it correctly or not

File: app/views/static_pages/help.html.erb (hunk 23 above)
Comment (nuriman1) [reply app/views/static_pages/help.html.erb]: done but can you check again if I do it correctly or not

File: app/views/static_pages/about.html.erb (hunk 22 above)
Comment (nuriman1) [reply app/views/static_pages/about.html.erb]: Honestly I am not very sure about indenting. But I changed the indent for those line, can you help me check and guide me if it is still wrong

**Comment by nuriman1** [comment]
ready

File: app/views/static_pages/about.html.erb [hunk 25]
```diff
@@ -0,0 +1,12 @@
+<%= provide :title, "About" %>
+
+<h1>About</h1>
+<p>
+  <%= link_to "<em>Ruby on Rails Tutorial</em>", "https://www.railstutorial.org/" %>, part of the
+    <%= link_to "Learn Enough", "https://www.learnenough.com/" %> family of tutorials, is a
+      <%= link_to "book", "https://www.railstutorial.org/book" %> and
+      <%= link_to "screencast series", "https://screencasts.railstutorial.org/" %> 
+      to teach web development with
+    <%= link_to "Ruby on Rails", "https://rubyonrails.org/" %>.
```
Comment (chungpv-1008) [inline app/views/static_pages/about.html.erb]: indent

File: app/views/static_pages/help.html.erb [hunk 26]
```diff
@@ -0,0 +1,10 @@
+<%= provide :title, "Help" %>
+
+<h1>Help</h1>
+<p>
+  Get help on the Ruby on Rails Tutorial at the 
+    <%= link_to "Rails Tutorial help page", Settings.links.rails_tutorial.help_page %>
+      
+    To get help on this sample app, see the
```
Comment (chungpv-1008) [inline app/views/static_pages/help.html.erb]: indent

**Review by chungpv-1008 — COMMENTED** [review]
merged

File: app/views/static_pages/about.html.erb (hunk 25 above)
Comment (nuriman1) [reply app/views/static_pages/about.html.erb]: done

File: app/views/static_pages/help.html.erb (hunk 26 above)
Comment (nuriman1) [reply app/views/static_pages/help.html.erb]: done

**Pull request merged**